src/components/cylc/TreeItem.vue

```diff
@@ -0,0 +1,194 @@
+        </v-flex>
+        <v-flex shrink>
+          <span class="mx-1">{{ node.name }}</span>
+        </v-flex>
+        <v-flex grow ml-4 v-if="!isExpanded">
+          <!-- Task summary -->
+          <job
+              v-for="(task, index) in node.children"
+              :key="`${task.id}-summary-${index}`"
+              :status="task.state" />
+        </v-flex>
+      </v-layout>
+      <v-layout @click="nodeClicked" row wrap v-else-if="node.__type === 'job'">
+        <v-flex shrink>
+          <job :status="node.state" />
+        </v-flex>
+        <v-flex xs2 md1 lg1>
+          <span class="mx-1">{{ node.name }}</span>
+        </v-flex>
+        <v-flex grow>
+          <span class="text-gray">{{ node.host }}</span>
+        </v-flex>
+      </v-layout>
+      <v-layout row wrap v-else>
+        <span @click="nodeClicked" class="mx-1">{{ node.name }}</span>
+      </v-layout>
+    </v-layout>
+    <span v-show="isExpanded">
+      <!-- component recursion -->
+      <TreeItem
+          v-for="child in node.children"
+          :key="child.id"
+          :node="child"
+          :depth="depth + 1"
+          :hoverable="hoverable"
+          :min-depth="minDepth"
+          :expanded="expanded"
+          v-on:tree-item-created="$emit('tree-item-created', $event)"
+          v-on:tree-item-expanded="$emit('tree-item-expanded', $event)"
+          v-on:tree-item-collapsed="$emit('tree-item-collapsed', $event)"
+          v-on:tree-item-clicked="$emit('tree-item-clicked', $event)"
+      ></TreeItem>
+    </span>
+  </div>
+</template>
+
+<script>
+import Task from '@/components/cylc/Task'
+import Job from '@/components/cylc/Job'
+
+export default {
+  name: 'TreeItem',
+  components: {
+    task: Task,
+    job: Job
+  },
+  props: {
+    node: {
+      type: Object,
+      required: true
+    },
+    depth: {
+      type: Number,
+      default: 0
+    },
+    minDepth: {
+      type: Number,
+      default: 0
+    },
+    hoverable: Boolean,
+    expanded: {
+      type: Boolean,
+      default: true
+    }
+  },
+  data () {
+    return {
+      active: false,
+      selected: false,
+      isExpanded: this.expanded
+    }
+  },
+  computed: {
+    hasChildren () {
+      return Object.prototype.hasOwnProperty.call(this.node, 'children')
+    }
+  },
+  created () {
+    this.$emit('tree-item-created', this)
+  },
+  methods: {
+    typeClicked () {
+      this.isExpanded = !this.isExpanded
+      if (this.isExpanded) {
+        this.$emit('tree-item-expanded', this)
+      } else {
+        this.$emit('tree-item-collapsed', this)
+      }
+    },
+    getTypeStyle () {
+      const styles = {}
+      if (this.hasChildren) {
+        styles['cursor'] = 'pointer'
+      }
+      return styles
+    },
+    nodeClicked (e) {
+      this.$emit('tree-item-clicked', this)
+      if (e.ctrlKey) {
+        console.log('Clicked with CTRL!')
+      } else {
+        console.log('Clicked without CTRL!')
+      }
```

Comment:
Good catch. This was actually for the selection, to choose one or multiple, but I didn't implement the complex case where you can select multiple nodes, between different heights/distances.

And actually, I implemented the ugly way (but the one I have in my memory). I believe Vue has another special way, something like `@click.ctrl=methodWithCtrl` which is the equivalent of `@click=method`, but only fires when the CTRL key is used too.

Better remove for now to avoid any confusion. Thanks :+1:

Reply:
Done.

Reply:
Removed one-too-many lines, and broke activating nodes. Thankfully we have some unit tests covering that :pray: and Travis told me I had to fix these. Now it should be good to go.